math_keyboard/lib/src/foundation/math2tex.dart

```diff
@@ -69,12 +68,26 @@ List<TeX> _convertToTeX(Expression mathExpression, TeXNode parent) {
         ),
       ];
     }
-    // Note that modulo is unsupported.
-    throw UnimplementedError();
+    if (result == null) {
+      // Note that modulo is unsupported.
+      throw UnimplementedError();
+    }
+    // Wrap with parentheses to keep precedence.
+    return [
+      TeXLeaf('('),
+      ...result,
+      TeXLeaf(')'),
+    ];
```

Comment:
So we now wrap with parentheses for all **operators**, correct? And it makes sense for all of them?

Reply:
@creativecreatorormaybenot Oh sorry forgot to write a comment about this. No it does **not** make sense. In the best case, we would just set parentheses where the creator puts them. Unfortunately, `math_expressions` sets parentheses everywhere when calling `toString()` on an `Expression`. This means, we cannot know which of them where originally set by the creator and we have to keep all of them to ensure we keep the precedence.

Reply:
@creativecreatorormaybenot We could check ourself, which parentheses are really necessary, but in this place it's not worth the effort, since this is only relevant for JARVIS and has no impact on end users. It would be more important to tackle this when showing the solution in the app🙂

Reply:
@creativecreatorormaybenot Also it does only apply for `binary` operators:)